src/net/sourceforge/kolmafia/KoLAdventure.java

```diff
@@ -454,6 +481,143 @@ private void validate1() {
     holidayAdventures.put("Drunken Stupor", null);
   }
 
+  // Validation part 0:
+
+  public boolean preValidateAdventure() {
+
+    if (this.zone.equals("Spring Break Beach")) {
+      // Unlimited adventuring if available
+      if (Preferences.getBoolean("sleazeAirportAlways")) {
+        return true;
+      }
+      if (!Preferences.getBoolean("_sleazeAirportToday")) {
+        // Take a look at the airport.
+        var request = new PlaceRequest("airport");
+        RequestThread.postRequest(request);
+      }
+      return Preferences.getBoolean("_sleazeAirportToday");
```

Comment:
All the airport versions are mostly the same, and could be refactored into `boolean checkAirportZone(String element)` to make the code read more easily.

You could go further and refactor everything but spacegate (and love tunnel, with its different comment) into `boolean checkZone(String alwaysPref, String todayPref, String place)`. I'm not sure whether this is better overall.

Reply:
checkZone works well and is clean.

Reply:
Done.